raft/raft.go

```diff
@@ -642,6 +648,14 @@ func stepLeader(r *raft, m pb.Message) {
 			r.id, r.raftLog.lastTerm(), r.raftLog.lastIndex(), r.Vote, m.From, m.LogTerm, m.Index, r.Term)
 		r.send(pb.Message{To: m.From, Type: pb.MsgVoteResp, Reject: true})
 		return
+	case pb.MsgReadIndex:
+		ri := None
+		if r.checkQuorum {
+			ri = r.raftLog.committed
+		}
```

Comment:
I think there's only two possible state here:
if r.checkQuorum is on, than r.raftLog.committed is exactly the value we want
any other cases, the readState value is uncertain.

so how about declaration a const variable called Uncertain in raft.go, so the code would be like:
ri := Uncertain
if r.checkQuorum {
...
}

so that we don't need a else clause since it already cover all the other cases, right now we reuse the variable None in raft.go